### DISCUSSION TITLE
[EssentialsDiscord] Add a getMember() method

### DISCUSSION THREAD

**Discussion by HyperCodec** [EssentialsX/Essentials#5033] [Ideas and Feature Suggestions] [answered]
There should be a way to get members in the server other than just using an event. Here are some ideas:

``` java

DiscordUtil.getMember(long id) // getId() should also return a long, as the id is a number 
DiscordUtil.getMember(String name)
// Could also be in a different class such as DiscordService. This is just an idea, so feel free to put it wherever you see fit

```

**Comment by pop4959** [comment]
In what way? As mentioned in #5032, EssentialsDiscord is not a general purpose API for Discord. You should be able to get members through any Discord wrapper of your choice (the EssentialsDiscord bot uses JDA, but this is an implementation detail).

**Reply by HyperCodec** [reply]
> In what way? As mentioned in #5032, EssentialsDiscord is not a general purpose API for Discord. You should be able to get members through any Discord wrapper of your choice (the EssentialsDiscord bot uses JDA, but this is an implementation detail).

A getMember() method would be useful for many things, such as if someone ran a command in-game. It could also be useful for things like saving data, as a user variable (or key in database or smth) could just be the id, which could then be used to fetch other information about the member temporarily. My point is, the getMember() method is pretty essential (pun unintended) to make bots, even if they aren't for general purposes.

**Reply by HyperCodec** [reply]
I actually have JDA directly imported in maven because for some reason it wasnt downloading the dependency from the essentialsxdiscord pom, so I can probably use that.

**Reply by JRoy** [reply]
I would disagree with pop here. I want our API to be very robust to allow for plugins to customize the experience without needing to use internals like JDA. A method like this will be added in the link addon

**Reply by HyperCodec** [reply]
> I would disagree with pop here. I want our API to be very robust to allow for plugins to customize the experience without needing to use internals like JDA. A method like this will be added in the link addon 

Thanks for agreeing with my suggestion. The link plugin seems to have a lot of the features that are missing. When do you think getMember() will be added to the link plugin and maybe later merged?

**Reply by pop4959** [reply]
I was under the impression that this was asking for the internal JDA Member object to be exposed. If that is not the case, then I think we're all on the same page here. Apologies for my confusion!

**Comment by JRoy** [comment, marked as answer]
DiscordService#getMemberById now exists.
